src/qt/sidechaindepositdialog.cpp

```diff
@@ -80,7 +82,7 @@ void SidechainDepositDialog::on_pushButtonDeposit_clicked()
     const CAmount& nValue = ui->payAmount->value();
     CTransactionRef tx;
     std::string strFail = "";
-    if (!pwalletMain->CreateSidechainDeposit(tx, strFail, nSidechain, nValue, keyID)) {
+    if (!vpwallets[0]->CreateSidechainDeposit(tx, strFail, nSidechain, nValue, keyID)) {
```

Comment:
I think we need ```if (!vpwallets.empty())``` first here, I'm new to the multi wallet support code though so we should double check.

Reply:
Same comment for other lines and from what I can tell everywhere in the upstream codebase is checking that the vector of wallets isn't empty before accessing. One of us can add those checks before we merge this :+1:

Reply:
Good point - I've added isempty checks for vpwallets